spark/src/main/scala/org/apache/comet/CometSparkSessionExtensions.scala

```diff
@@ -189,6 +189,22 @@ class CometSparkSessionExtensions
             }
 
           // data source V1
+          case scanExec @ FileSourceScanExec(
+                HadoopFsRelation(_, partitionSchema, _, _, _: ParquetFileFormat, _),
+                _: Seq[_],
+                requiredSchema,
+                _,
+                _,
+                _,
+                _,
+                _,
+                _)
+              if CometNativeScanExec.isSchemaSupported(requiredSchema)
```

Comment:
we could also fall back for bucketed scans and other unsupported cases (can be a separate PR)